src/crystal/project/dsl.cr

```diff
@@ -13,5 +13,9 @@ struct Crystal::Project::DSL
     def github(repository, name = nil : String)
       @project.dependencies << GitHubDependency.new(repository, name)
     end
+
+    def local(path, name = nil : String)
```

Comment:
If you can send another pull request, please do it :)

You can also try to implement what @zamith said below:

```
What about moving this unless and case to the Dependency class? Apart from variable names it looks the same for local and github, and if this changes for some reason, we can then just change it in one place.
```

I think that logic was in a subclass because we only had one subclass, but now it makes sense to move that logic up.